### ISSUE TITLE
Uikit 3

### ISSUE THREAD

**Issue by Mostafa20946** [Getbeans/Beans#45]
Hello
Please update the theme to **uikit 3**.
Thank you.

**Comment by ThierryA** [comment]
Hi @Mostafa20946, kindly refer to [this thread](https://community.getbeans.io/discussion/uikit-3-2/) on the forum.

Thanks,

**Comment by hellofromtonya** [comment]
@Mostafa20946 - We're excited about UIKit 3, me especially as I work in Sass and not LESS.  At the moment, it's still in beta, though it is moving along in the development stages.  

Our plan is to integrate it into the Beans v2 Roadmap.

**Comment by alexadark** [comment]
I think we should integrate uikit3 as soon as possible, i
I am using it since June (in my genesis child theme), without any problem.
I chat with people here https://gitter.im/uikit/uikit, and everybody is with v3
Things are stable, but some components are missing, especially the dynamic grid, so I use isotope in between
The thing is that we cannot begin to build child themes or plugins since we aren't on uikit3 because we will need to rebuild them after.
Anyway i think it will be ready soon!

**Comment by paaljoachim** [comment]
I came across this blog post: https://yootheme.com/blog/2017/10/20/uikit-3-beta-31-released

Closer to the bottom of the article the following is mentioned:

What's Next?

Now the Slider, the Slideset and the Dynamic Grid are the only three components which stand between us and the stable version of UIkit 3. We are sure that you will love our new release! Leave your feedback in the comments section and let us know what you would like to see in future.

--

So it looks like it is real close to a stable release of Uikit 3.

**Comment by hellofromtonya** [comment]
@paaljoachim @alexadark @Mostafa20946 UIKit 3 is really close to being a stable release.  Thank you @paaljoachim for sharing the article.  We want to stay on top of this!

## Beans 2 - "You Pick" what you want.

As part of Beans 2, we are planning to make  Beans UI Framework Agnostic, see #51.  What does that mean?  

It means you will be able to select which UI framework you want for your project.  You could use UIKit 2, UIKit 3, Zurb's Foundation, Bootstrap, your own, or whatever.  You pick.

UIKit has been part of Beans from the beginning.  But we are moving into the future and want to make Beans even more extensible.  Framework Agnostic as part of the "You Pick" architecture is a key promise for us.

## What does that mean for UIKit?
We have decisions to make, which is why this open issue is important and vital to Beans 2 Roadmap:

1. Do we keep UIKit 2 integrated into Beans? Or make it a separate plugin?
2. Do we integrate UIKit 3 into Beans when it's stable? Or does it become a separate plugin?

These are decisions yet to be made.

Bottom line: UIKit will be available, both versions.  Then you can pick and choose what you need per web page on each project for maximum flexibility.

**Comment by vayurobins** [comment]
Making Beans UI Framework Agnostic would be awesome! So for instance, if Uikit 3 stays in Beta for a long time, then people can choose to use version 2 or 3 as they prefer. Like many, I also really look forward to starting using Uikit 3, even though still in beta.
